geniza/footnotes/models.py

```diff
@@ -3,55 +3,92 @@
 from django.contrib.contenttypes.fields import GenericForeignKey
 
 from multiselectfield import MultiSelectField
-# https://pypi.org/project/django-multiselectfield/
-from sortedm2m.fields import SortedManyToManyField
-# https://github.com/jazzband/django-sortedm2m
+
 
 class SourceType(models.Model):
+    '''type of source'''
     type = models.CharField(max_length=255)
 
     def __str__(self):
         return self.type
 
 
+class SourceLanguage(models.Model):
+    '''language of a source document'''
+    name = models.CharField(max_length=255)
+    code = models.CharField(max_length=10, help_text='ISO language code')
+
+    def __str__(self):
+        return self.name
+
+
 class Creator(models.Model):
+    '''author or other contributor to a source'''
     first_name = models.CharField(max_length=255)
     last_name = models.CharField(max_length=255)
 
     def __str__(self):
         return f'{self.last_name}, {self.first_name}'
 
     class Meta:
-        ordering = ['last_name']
+        ordering = ['last_name', 'first_name']
+        constraints = [
+            models.UniqueConstraint(fields=['first_name', 'last_name'],
+                                    name='creator_unique_name')
+        ]
+
+
+class Authorship(models.Model):
+    """Ordered relationship between :class:`Creator` and :class:`Source`."""
+    creator = models.ForeignKey(Creator, on_delete=models.CASCADE)
+    source = models.ForeignKey('Source', on_delete=models.CASCADE)
+    sort_order = models.PositiveSmallIntegerField(default=0)
```

Comment:
Just to make sure I understand – this through-model creates an ordered relationship, but it also implies a nature to the relationship, right? It's only for authors, otherwise it'd be called `Creatorship`. And similarly, if we had editors or other kinds of creators, we'd need separate through-models for associating them, since there's no concept of "creator type" right now.

Reply:
I named it "creatorship" at first but that sounded terrible! (is that even a real word?) I didn't intend to imply that this was for authors only. Should I rename it back? 

Or should we ask what kind of creators the team cares about, and whether we should differentiate them? (🤔 They might only care about authors right now; and generating a citation properly requires we know which role a creator has...)

Reply:
I looked at the user story again, and they explicitly requested author. This seems like a reasonable solution, doesn't it? — generic creators, but only define authorship for now and we can create more through models if there are other creator types we care about

Reply:
Sounds good to me!